### DISCUSSION TITLE
Would be great to rely on nio package

### DISCUSSION THREAD

**Discussion by avianey** [ngageoint/tiff-java#45] [Ideas]
To avoid loading the whole file into memory as a byte buffer, it could be great to move to the java nio package, using [FileChannel](https://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileChannel.html). Some of the advantages would be :

- speed improvement (no need to load everything into memory)
- less risk of OOME (specially on the Android side)
- some operation don't need to read the whole file/raster and only requires a direct / single pixel access such as reading an elevation value from a GeoTIFF DEM model 

on the other hand, it could be great to keep current implementation for use case that requires :

- reading a tiff from something else than a file
- access multiple time a same information (multiple read of the same pixel of the same tiff)

What do you think ?

**Comment by bosborn** [comment]
I am not opposed if the current functionality is preserved.
No current plans, but open to pull requests.
